Assets/Script/Player/PlayerManager.cs

```diff
@@ -1,180 +1,216 @@
+                            _rigidbody2D.velocity += Vector2.up;
+                        }
+                    }
+                    else if (!GetJumptKey() && playerData.jumpData.nowForceTime != 0)
+                    {
+                        JumpEnd();
+                    }
+
+                    break;
+                }
+                case false when playerData.jumpData.CdCounter > playerData.jumpData.Cd:
+                    playerData.stateCheck.Jump = true;
+                    playerData.jumpData.CdCounter = 0;
+                    break;
+                case false:
+                    playerData.jumpData.CdCounter += Time.deltaTime;
+                    break;
+            }
         }
-        else
+
+        private void JumpEnd()
         {
-            Move(0.0f, Rigidbody.velocity.y);
+            playerData.jumpData.nowForceTime = 0;
+            playerData.stateCheck.Jump = false;
+            onGround = false;
         }
 
-        // will do keep a time and  no work
-        if (Input.GetKey("w") && _touchGround)
+        private bool GetJumptKey()
         {
-            Move(Rigidbody.velocity.x, data.jumpSpeed);
-            _touchGround = false;
+            return Input.GetKey(KeyCode.Space) || Input.GetKey(KeyCode.W);
         }
-    }
 
-    private void Move(Vector2 unitMove)
-    {
-        transform.position = (Vector2)transform.position + unitMove;
-    }
+        private Vector2 HorizontalMoveManager()
+        {
+            if (GetRightKey() && GetLeftKey())
+            {
+                return Vector2.zero;
+            }
+            else if (GetRightKey())
+            {
+                return new Vector2(nowSpeed, _rigidbody2D.velocity.y);
+            }
+            else if (GetLeftKey())
+            {
+                return new Vector2(-nowSpeed, _rigidbody2D.velocity.y);
+            }
 
-    private void Move(float x, float y)
-    {
-        Rigidbody.velocity = new Vector2(x, y);
-        //
-    }
+            return new Vector2(0, _rigidbody2D.velocity.y);
+        }
 
-    public void NewGame()
-    {
-        gameObject.transform.Translate(0.0f, 0.0f, 0.0f);
-        data.pastLocal.Clear();
-        data.queueTime = 0;
-        _touchGround = false;
-        attacked = false;
-    }
+        private bool GetLeftKey()
+        {
+            return Input.GetKey(KeyCode.A) || Input.GetKey(KeyCode.LeftArrow);
+        }
 
-    private void OnTriggerEnter2D(Collider2D collision)
-    {
-        Debug.Log(collision.gameObject.tag);
-        if (attacked == false
-            && collision.gameObject.CompareTag("enemy")
-            && collision.gameObject.transform.rotation == this.transform.rotation)
+        private bool GetRightKey()
         {
-            data.enemyHp -= data.playerDamege;
-            attacked = true;
-            if (data.enemyHp <= 0)
-            {
-            }
+            return Input.GetKey(KeyCode.D) || Input.GetKey(KeyCode.RightArrow);
         }
```

Comment:
I prefer use variable instead to function if no other references

Reply:
ok